libs/virtual-node/cpi-datamodel/src/integrationTest/kotlin/net/corda/libs/cpi/datamodel/entities/tests/CpiMetadataRepositoryImplTest.kt

```diff
@@ -0,0 +1,72 @@
+package net.corda.libs.cpi.datamodel.entities.tests
+
+import net.corda.crypto.core.SecureHashImpl
+import net.corda.db.admin.impl.ClassloaderChangeLog
+import net.corda.db.admin.impl.LiquibaseSchemaMigratorImpl
+import net.corda.db.schema.DbSchema
+import net.corda.db.testkit.DbUtils
+import net.corda.libs.cpi.datamodel.CpiEntities
+import net.corda.libs.cpi.datamodel.repository.factory.CpiCpkRepositoryFactory
+import net.corda.libs.packaging.core.CpiIdentifier
+import net.corda.orm.EntityManagerConfiguration
+import net.corda.orm.impl.EntityManagerFactoryFactoryImpl
+import net.corda.orm.utils.transaction
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CpiMetadataRepositoryImplTest {
+
+    private val dbConfig: EntityManagerConfiguration = DbUtils.getEntityManagerConfiguration("cpk_file_db")
+    private val emf = EntityManagerFactoryFactoryImpl().create(
+        "test_unit",
+        CpiEntities.classes.toList(),
+        dbConfig
+    )
+
+    private val cpiMetadataRepository = CpiCpkRepositoryFactory().createCpiMetadataRepository()
+
+    init {
+        val dbChange = ClassloaderChangeLog(
+            linkedSetOf(
+                ClassloaderChangeLog.ChangeLogResourceFiles(
+                    DbSchema::class.java.packageName,
+                    listOf("net/corda/db/schema/config/db.changelog-master.xml"),
+                    DbSchema::class.java.classLoader
+                )
+            )
+        )
+        dbConfig.dataSource.connection.use { connection ->
+            LiquibaseSchemaMigratorImpl().updateDb(connection, dbChange)
+        }
+    }
+
+    @AfterAll
+    fun cleanUp() {
+        emf.close()
+    }
+
+    @Test
+    fun `findAll stores CPI MetaData without any cpks successfully`(){
```

Comment:
```suggestion
    fun `findAll finds CPI metadata without any CPKs successfully`() {
```
Some of this is probably a bit nitpicky